packages/atlas/src/views/viewer/NftPurchaseView/NftPurchaseView.tsx

```diff
@@ -0,0 +1,377 @@
+  useEffect(() => {
+    if (type === 'buy_now') {
+      return
+    }
+    const subscription = watch(({ bid }) => {
+      if (bid >= FIXED_PRICE) {
+        setBuyNowInfo(true)
+      } else {
+        setBuyNowInfo(false)
+      }
+      if (bid > FIXED_PRICE) {
+        setValue('bid', FIXED_PRICE.toString())
+      }
+    })
+
+    return () => subscription.unsubscribe()
+  }, [setValue, type, watch])
+
+  const handleCloseClick = () => setIsNftPurchaseOpen(false)
+  const handleSubmit = useCallback(() => {
+    return createSubmitHandler((_) => null)
+  }, [createSubmitHandler])
+
+  const placedBid = getValues().bid
+  const timeLeftUnderMinute = timeLeftSeconds && timeLeftSeconds < 60
+  const insufficientFoundsError = errors.bid && errors.bid.type === 'bidTooHigh'
+  const primaryButtonText = type === 'buy_now' || placedBid >= FIXED_PRICE ? 'Buy NFT' : 'Place bid'
+
+  return (
+    <CSSTransition
+      in={isNftPurchaseOpen}
+      appear
+      mountOnEnter
+      unmountOnExit
+      timeout={{ enter: 0, exit: parseInt(cVar('animationTimingSlow', true)) }}
+      classNames="place-bid"
+    >
+      <Container role="dialog">
+        {/* this needs to be removed after integration */}
+        <div style={{ position: 'fixed', top: '10px', left: '10px' }}>
+          <Button variant="tertiary" onClick={() => setType('auction')}>
+            auction
+          </Button>
+          <Button variant="tertiary" onClick={() => setType('auction')}>
+            open auction
+          </Button>
+          <Button variant="tertiary" onClick={() => setType('buy_now')}>
+            buy now
+          </Button>
+        </div>
+        <DrawerHeader onCloseClick={handleCloseClick} />
+        <Content>
+          <NftPreview>
+            <NftCard title="title" {...DUMMY_NFT_TILE_PROPS} fullWidth={!mdMatch} />
+          </NftPreview>
+          <PlaceBidWrapper>
+            <InnerContainer>
+              <Header>
+                <Text variant="h600">{type === 'auction' ? 'Place a bid' : 'Buy NFT'}</Text>
+                {type !== 'open_auction' && (
+                  <FlexWrapper>
+                    <EndingTime>
+                      <Text variant="h300" secondary>
+                        Ending in:
+                      </Text>
+                      <Timer
+                        variant="h200"
+                        spacing={{ left: 4, right: 2 }}
+                        color={timeLeftUnderMinute ? cVar('colorTextError', true) : undefined}
+                      >
+                        {timeLeftSeconds &&
+                          (!timeLeftUnderMinute ? formatDurationShort(timeLeftSeconds, true) : 'Under 1 min')}
+                      </Timer>
+                    </EndingTime>
+                    <FlexWrapper>
+                      <Text variant="t100" secondary spacing={{ left: 2, right: 1 }}>
+                        {convertMsTimestampToBlock(END_TIME)} blocks
```

Comment:
This is supposed to show how many blocks are left till the auction end. Currently, you show ending block estimate based on the end time. This value should be dynamically updated as time passes